torchvision/datasets/svhn.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import print_function
+import scipy.io as sio
+import torch.utils.data as data
+from PIL import Image
+import os
+import os.path
+import errno
+import numpy as np
+import sys
+
+
+class SVHN(data.Dataset):
+    url = ""
+    filename = ""
+    file_md5 = ""
+
+    split_list = {
+        'train': ["http://ufldl.stanford.edu/housenumbers/train_32x32.mat",
+                  "train_32x32.mat", "e26dedcc434d2e4c54c9b2d4a06d8373"],
+        'test': ["http://ufldl.stanford.edu/housenumbers/test_32x32.mat",
+                 "test_32x32.mat", "eb5a983be6a315427106f1b164d9cef3"],
+        'extra': ["http://ufldl.stanford.edu/housenumbers/extra_32x32.mat",
+                  "extra_32x32.mat", "a93ce644f1a588dc4d68dda5feec44a7"]}
+
+    def __init__(self, root, split='train', transform=None, target_transform=None, download=False):
+        self.root = root
+        self.transform = transform
+        self.target_transform = target_transform
+        self.split = split  # training set or test set or extra set
+
+        if self.split in self.split_list:
+            self.url = self.split_list[split][0]
+            self.filename = self.split_list[split][1]
+            self.file_md5 = self.split_list[split][2]
+
+            if download:
+                self.download()
+
+            if not self._check_integrity():
+                    raise RuntimeError('Dataset not found or corrupted.' +
+                                       ' You can use download=True to download it')
+
+            # reading(loading) mat file as array
+            loaded_mat = sio.loadmat(os.path.join(root, self.filename))
+
+            if self.split != 'test':
+                self.train_data = loaded_mat['X']
+                self.train_labels = loaded_mat['y']
+                self.train_data = np.transpose(self.train_data, (3, 2, 1, 0))
+            else:
+                self.test_data = loaded_mat['X']
+                self.test_labels = loaded_mat['y']
+                self.test_data = np.transpose(self.test_data, (3, 2, 1, 0))
+        else:
+            print ("Wrong dataset entered! Please use split=train or split=extra or split=test")
```

Comment:
It might be better to raise an error in here.
Maybe something like
```python
raise ValueError("Wrong dataset entered! Please use split=train, split=extra or split=test")
```